src/app/stripe.service.ts

```diff
@@ -391,4 +361,42 @@ export class StripeService {
       });
     });
   }
+
+  async confirmPaymentWithPaymentElement(donation: Donation, elements: StripeElements): Promise<PaymentIntentResult | undefined>
+  {
+    const {error: submitError} = await elements.submit();
+    if (submitError) {
+      console.error(submitError); // @todo handle this error better.
+      return;
+    }
+
+    // const donorName = donation.firstName || '' + ' ' +  donation.lastName;
+
+    // If we want to not show billing details inside the Stripe payment element we have to pass billing details
+    // as payment_method_data here, with at least this much detail - but we don't collect addresses in that much detail.
+    // const paymentMethodData = {
+    //   billing_details:
+    //     {
+    //       name: donorName,
+    //       address: {
+    //         city: '',
+    //         state: '',
+    //         country: donation.countryCode,
+    //         postal_code: donation.billingPostalAddress
+    //       },
+    //       email: donation.emailAddress,
+    //       phone: ''
+    //     }
+    // };
+
+    return await this.stripe?.confirmPayment({
+      elements: elements,
+      clientSecret: donation.clientSecret as string,
+      redirect: 'if_required',
```

Comment:
I'm a bit nervous about the possibility that we could end up with different, regional payment method types enabled on the dashboard that we have never tested or specifically coded for.

It seems like maybe this is the only way the new element can be used though – is that your understanding @bdsl?

Have we done any testing of a method that redirects if so?

Reply:
Yes I think this is what the new element makes us do - butting redirect: 'never' would be a type error: 
![image](https://github.com/thebiggive/donate-frontend/assets/159481/aaea891d-6c93-49cb-91ed-ef2ffb8537db)